### ISSUE TITLE
Tag push event not working with workflow plugin

### ISSUE THREAD

**Issue by peternijssen** [jenkinsci/gitlab-plugin#167]
I have the workflow plugin installed in Jenkins and I am running Gitlab 8.3. I've setup a web hook which notifies Jenkins as soon as a new tag is pushed. This works fine with a freestyle project, but not with a workflow project. I think because the SCM is missing from the settings in a workflow project.

This is the log for a push to a workflow project;

```
Dec 23, 2015 12:21:36 PM INFO com.dabsquared.gitlabjenkins.GitLabWebHook getDynamic
WebHook called with url: 
Dec 23, 2015 12:21:36 PM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$2 run
Zipper triggered for push.
Dec 23, 2015 12:21:36 PM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$2 createActions
GitLab Push Request from branch refs/tags/1.0.0.
Dec 23, 2015 12:21:36 PM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$2 createActions
Trying to get name and URL for job: Zipper
Dec 23, 2015 12:21:36 PM WARNING com.dabsquared.gitlabjenkins.GitLabPushTrigger$DescriptorImpl getSourceRepoNameDefault
Could not find GitSCM for project. Project = 4, next build = {2}
```

This is for the freestyle project

```
Dec 23, 2015 12:22:00 PM INFO com.dabsquared.gitlabjenkins.GitLabWebHook getDynamic
WebHook called with url: 
Dec 23, 2015 12:22:00 PM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$2 run
Zipper2 triggered for push.
Dec 23, 2015 12:22:00 PM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$2 createActions
GitLab Push Request from branch refs/tags/1.0.0.
Dec 23, 2015 12:22:00 PM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$2 createActions
Trying to get name and URL for job: Zipper2
Dec 23, 2015 12:22:02 PM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$2 run
GitLab Push Request detected in Zipper2. Triggering  #4
Dec 23, 2015 12:22:10 PM INFO hudson.model.Run execute
Zipper2 #4 main build action completed: SUCCESS
```

The pull request that added compliance for the workflow project; #100

**Comment by omehegan** [comment]
Yep, I helped work on that PR, but I now know that the support for Workflow is incomplete and broken. I have it partly fixed but need to do some more work. When I get back from vacation I'll try to finish it.

**Comment by peternijssen** [comment]
Great! Let me know if I can help testing.

**Comment by peternijssen** [comment]
@omehegan any news on this? Much appreciated!

**Comment by ehaselwanter** [comment]
@omehegan would be interested too :-)

**Comment by gtheraud** [comment]
Hi @omehegan, this feature would help. as @peternijssen said, if you need support for testing, I can help as well.

**Comment by omehegan** [comment]
Sorry for the long silence! I am back to working on the plugin again this week, and will be trying to revive the fix for Workflow support.

**Comment by peternijssen** [comment]
You are awesome @omehegan!

**Comment by omehegan** [comment]
Have any of you gotten GitLab to trigger a Workflow job from a _merge request?_ I have it triggering the job when I push, but when I create an MR it does not trigger. I can't figure out why. I'm wondering if it's related to the tag issue.

**Comment by peternijssen** [comment]
The "annoying" part is that there are several ways.
1. A web hook in Gitlab
2. The new Jenkins service in Gitlab
3. The old Jenkins service in Gitlab

I'll try the first 2 methods today and let you know the results.

**Comment by ehaselwanter** [comment]
yes. with a webhook

```
Feb 17 07:50:10 jenkins-poc [jenkins]: INFO: pipeline-test triggered for merge request.
Feb 17 07:50:10 jenkins-poc [jenkins]: Feb 17, 2016 7:50:10 AM com.dabsquared.gitlabjenkins.GitLabPushTrigger$3 createActions
Feb 17 07:50:10 jenkins-poc [jenkins]: INFO: Trying to get name and URL for job: pipeline-test
Feb 17 07:50:10 jenkins-poc [jenkins]: Feb 17, 2016 7:50:10 AM com.dabsquared.gitlabjenkins.GitLabPushTrigger$DescriptorImpl getSourceRepoNameDefault
Feb 17 07:50:10 jenkins-poc [jenkins]: WARNING: Could not find GitSCM for project. Project = 20, next build = {2}
```

```
java -jar  /var/lib/jenkins/war/WEB-INF/jenkins-cli.jar -s http://localhost:8080 version
1.617
```

```
ace-editor=1.0.1
analysis-core=1.75
ansicolor=0.4.1
ant=1.2
antisamy-markup-formatter=1.1
artifactory=2.2.7
aws-java-sdk=1.10.45
backup=1.6.1
build-monitor-plugin=1.6+build.138
build-name-setter=1.3
build-pipeline-plugin=1.4.5
checkstyle=3.41
cloudbees-folder=5.1
clover=4.4.0
cobertura=1.9.6
console-column-plugin=1.5
copyartifact=1.37
credentials=1.24
cucumber-reports=0.0.23
cucumber-testresult-plugin=0.7
cvs=2.11
dashboard-view=2.9.4
delivery-pipeline-plugin=0.9.8
durable-task=1.7
embeddable-build-status=1.9
envfile=1.2
envinject=1.90
external-monitor-job=1.4
extra-columns=1.14
git=2.4.2
git-client=1.19.4
gitlab-plugin=1.1.28
git-server=1.6
greenballs=1.15
htmlpublisher=1.3
jackson2-api=2.5.4
javadoc=1.1
jquery=1.11.2-0
jquery-detached=1.2
junit=1.10
ldap=1.11
log-parser=1.0.8
mailer=1.16
mapdb-api=1.0.6.0
matrix-auth=1.2
matrix-project=1.6
maven-plugin=2.7.1
multi-branch-project-plugin=0.4.1
pam-auth=1.1
parameterized-trigger=2.30
ruby-runtime=0.12
Running plugin enumerator
rvm=0.4
s3=0.8
scm-api=1.0
scriptler=2.7
script-security=1.17
slack=1.8.1
ssh=2.4
ssh-credentials=1.11
ssh-slaves=1.9
subversion=2.5.7
tap=1.20
thinBackup=1.7.4
token-macro=1.12.1
Total number of plugins: 77
translation=1.10
windows-slaves=1.0
workflow-aggregator=1.13
workflow-api=1.13
workflow-basic-steps=1.13
workflow-cps=1.13
workflow-cps-global-lib=1.13
workflow-durable-task-step=1.13
workflow-job=1.13
workflow-scm-step=1.13
workflow-step-api=1.13
workflow-support=1.13
xvfb=1.0.16
```

**Comment by peternijssen** [comment]
@omehegan I noticed in a previous commit you worked on the workflow integration. Should everything currently now work or are there still parts which are missing? Then I know what to report and what not to report :)

Thanks for your work so far! Really appreciated.

**Comment by omehegan** [comment]
@peternijssen everything should work now with the latest version of the plugin, .32. The one caveat is that if you don't run the job manually once and have it succeed, the fact that it is a GitSCM job will not be stored, and the GitLab plugin will ignore it during triggers, showing the warning you noted in your original issue. We are going to remove this restriction so that the plugin will just blindly trigger the job you specify, regardless of whether it thinks it is using Git or not. But for now that small issue remains.

Also someone in IRC found an issue which seems to be related to the Folders plugin, but his jobs are also Workflow so there could be overlap. I'm going to look into that once I have more info to try and reproduce.

**Comment by peternijssen** [comment]
Thanks! I am currently running the Jenkins 2.0 BETA and I am using the Folders plugin. I'll let you know which issues arise at my side. Any things I can configure in Jenkins to give you good error reports? (If any occurs of course :))

**Comment by omehegan** [comment]
Any update on this issue?

**Comment by peternijssen** [comment]
"Push Events" and "Tag Push Events" work great! So far as I am aware off "Merge Request Events" also works, but I am unable to do a thorough check because I am blocked by an issue in the pipeline plugin itself;  https://issues.jenkins-ci.org/browse/JENKINS-33719

**Comment by omehegan** [comment]
Yes, I'm aware of that issue with the Pipeline plugin. Ok, I will close this. Thanks!

**Comment by gtheraud** [comment]
@peternijssen : I did not try "Tag Push Events" so far with Jenkins Pipeline.
Does it require Beta2 ? Does it mean that, if I push a tag on the git repo (and not necessarily on the HEAD), it could be possible to get information like "BUILD_CAUSE = tag push" or even the labels pushed ?

**Comment by smutel** [comment]
Hello,

Do you know if this feature should work now ?
In gitlab, I seteup the integration on Tag push events.
In jenkins, in the tag tab, nothing is launched automatically ...

Thanks.

**Comment by omehegan** [comment]
@smutel please open a new issue if you are still having this problem. Include relevant logs and screenshots (I don't know what you mean by the tag tab).
